### PR TITLE
Add NeoX T4 testnet chain

### PR DESCRIPTION
**Pull request by joeqian10**

### CODE REVIEW

File: chains/neoxt4/logo_x.svg [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+</g>
+</svg>
```
Comment (paulbalaji) [inline chains/neoxt4/logo_x.svg:8]: ran through svg minify https://jakearchibald.github.io/svgomg/
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" width="2010" height="2010" fill="none"><mask id="a" width="2010" height="2010" x="0" y="0" maskUnits="userSpaceOnUse" style="mask-type:alpha"><path fill="#D9D9D9" d="M0 0h2010v2010H0z"/></mask><g mask="url(#a)"><path fill="#00E599" d="M1724.36 239.714 1574.2 85.21c-73-74.961-195.63-68.298-259.86 14.16L880.542 657.416 565.206 253.874c-65.904-84.124-198.546-66.633-239.84 32.483l-75.08 179.907c-22.941 54.972-15.85 117.856 18.77 166.165l307.829 431.031-271.957 364.81c-40.46 54.14-42.546 127.43-6.257 184.07l81.754 126.19c59.23 91.2 189.786 99.11 259.443 16.24l235.668-280.69 321.174 433.11c50.47 67.88 143.07 89.54 218.15 50.39l169.35-87.87c95.93-49.98 122.63-174.91 55.48-259.45l-421.7-532.64 514.29-607.191c57.15-67.465 53.81-167.414-7.92-230.715Zm-65.9 168.247-554.35 654.249c-6.67 7.91-6.67 19.15-.41 27.07l460.07 580.53c29.2 37.07 17.52 91.62-24.19 113.69l-169.35 87.87c-32.95 17.08-73.41 7.92-95.52-22.07l-377.901-509.74c-8.342-11.24-25.027-11.66-33.786-.83L566.04 1692.72c-27.947 33.31-80.086 29.98-103.861-6.25l-81.754-126.19c-14.599-22.48-13.765-52.05 2.503-73.71l304.908-408.95c5.423-7.5 5.84-17.49.417-25.41L348.724 576.624c-15.016-21.239-18.352-48.725-8.342-72.463l75.08-179.907c12.514-30.401 53.391-35.815 73.829-9.995l374.566 478.92a21.557 21.557 0 0 0 34.204 0l493.029-634.673c27.94-36.231 81.75-39.146 113.87-6.247l150.16 154.504c26.69 27.902 28.36 71.63 3.34 101.198Z"/></g></svg>
```

File: chains/neoxt4/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# yaml-language-server: $schema=../schema.json
+blockExplorers:
+  - apiUrl: https://xt4scan.ngd.network
```
Comment (paulbalaji) [inline chains/neoxt4/metadata.yaml:3]:
```suggestion
  - apiUrl: https://xt4scan.ngd.network/api
```

File: chains/neoxt4/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    name: xt4scan
+    url: https://xt4scan.ngd.network
+blocks:
+  confirmations: 1
+  estimateBlockTime: 15
+  reorgPeriod: 12
+chainId: 12227332
+deployer:
+  name: Neo Global Development
+  url: https://x.neo.org/
```
Comment (paulbalaji) [inline chains/neoxt4/metadata.yaml:14]:
```suggestion
  url: https://x.neo.org
```

File: chains/neoxt4/logo_x.svg (hunk 1 above)
Comment (joeqian10) [reply chains/neoxt4/logo_x.svg]: Fixed in dd41a59e930d91a34b54bb96f3fb58fcfc76b8c0

File: chains/neoxt4/metadata.yaml (hunk 2 above)
Comment (joeqian10) [reply chains/neoxt4/metadata.yaml]: Modified in dd41a59e930d91a34b54bb96f3fb58fcfc76b8c0.

File: chains/neoxt4/metadata.yaml (hunk 3 above)
Comment (joeqian10) [reply chains/neoxt4/metadata.yaml]: Modified in dd41a59e930d91a34b54bb96f3fb58fcfc76b8c0.

File: chains/neoxt4/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+domainId: 12227332
+isTestnet: true
+name: neoxt4
+nativeToken:
+  decimals: 18
+  name: GAS
+  symbol: GAS
+protocol: ethereum
+rpcUrls:
+  - http: https://neoxt4seed1.ngd.network
```
Comment (paulbalaji) [inline chains/neoxt4/metadata.yaml:25]: adding another fallback RPC
```suggestion
  - http: https://neoxt4seed1.ngd.network
  - http: https://testnet.rpc.banelabs.org
```

**Review by paulbalaji — COMMENTED** [review]
LGTM with one nit

**Review by jmrossy — APPROVED** [review]

**Pull request merged**